isis/src/base/objs/ProcessRubberSheet/ProcessRubberSheet.cpp

```diff
@@ -1075,22 +1078,43 @@ namespace Isis {
         InputCubes[0]->read(iportal);
         double dn = interp.Interpolate(isamp, iline, iportal.DoubleBuffer());
 
-        // Write the output value at the appropriate location
-        if (dn != Null) {
-          obrick.SetBasePosition(osamp,oline,iportal.Band());
-          obrick[0] = dn;
-          OutputCubes[0]->write(obrick);
+        // Write the output value at the appropriate location in buffer
+        if (dn == Null) {
+          foundNull = true;
         }
+        oBrick[brickIndex] = dn;
+        brickIndex++;
       }
     }
+
+    // If there are any special pixel Null values in this output brick, we may be
+    // up against an edge of the input image where the interpolaters get Nulls from 
+    // outside the image. Since the patches have some overlap due to finding the 
+    // rectangular area (bounding box, min/max line/samp) of the four corner points
+    // projected into the output space and then squaring that 
+    // possibly rotated patch in the output space by finding the min/max line/samp. 
+    // This overlap causes valid DNs from a previously processed patch to be replaced with 
+    // Null DNs from this patch.
+    if (foundNull) {
+     Brick readBrick(*OutputCubes[0], osampMax-osampMin+1, olineMax-olineMin+1, 1);
+     readBrick.SetBasePosition(osampMin, olineMin, iportal.Band());
+     OutputCubes[0]->read(readBrick);
+     for (brickIndex = 0; brickIndex<(osampMax-osampMin+1)*(olineMax-olineMin+1); brickIndex++) {
+       if (readBrick[brickIndex] != Null) {
+         oBrick[brickIndex] = readBrick[brickIndex];
+       }
+     }
+    }
```

Comment:
Instead of having to check if for foundNull and then re-process, you could always start out by reading into oBrick and then if the interpolator returns a null, not assigning over that pixel in oBrick. This way, you don't overwrite anything with Nulls. It'll be a little bit more costly to read every patch, but I'm not sure if that is significant. May be worth pushing through a profielr.

Reply:
I'm actually in the process of doing just that. It turns out the output patches can have a lot of overlap. Without this current logic the edges of the projected area are a lot rougher. This also slows it down more than I like, but the output is nearly identical to before our changes.